src/lazy/encoder/writer.rs

```diff
@@ -449,47 +437,62 @@ impl<'value, V: ValueWriter> ValueWriter for ApplicationValueWriter<'value, V> {
     fn list_writer(self) -> IonResult<Self::ListWriter> {
         Ok(ApplicationListWriter::new(
             self.encoding,
+            self.value_writer_config,
             self.raw_value_writer.list_writer()?,
         ))
     }
 
     fn sexp_writer(self) -> IonResult<Self::SExpWriter> {
         Ok(ApplicationSExpWriter::new(
             self.encoding,
+            self.value_writer_config,
             self.raw_value_writer.sexp_writer()?,
         ))
     }
 
     fn struct_writer(self) -> IonResult<Self::StructWriter> {
+        let config = self.value_writer_config;
         Ok(ApplicationStructWriter::new(
             self.encoding,
+            config,
             self.raw_value_writer.struct_writer()?,
         ))
     }
 
     fn eexp_writer<'a>(self, macro_id: impl Into<MacroIdRef<'a>>) -> IonResult<Self::EExpWriter> {
         Ok(ApplicationEExpWriter::new(
             self.encoding,
+            self.value_writer_config,
             self.raw_value_writer.eexp_writer(macro_id)?,
         ))
     }
 }
 
 pub struct ApplicationStructWriter<'value, V: ValueWriter> {
-    encoding: &'value mut WriteContext,
+    encoding: &'value mut WriterContext,
     raw_struct_writer: V::StructWriter,
+    value_writer_config: ValueWriterConfig,
 }
 
 impl<'value, V: ValueWriter> ApplicationStructWriter<'value, V> {
     pub(crate) fn new(
-        encoding_context: &'value mut WriteContext,
+        encoding_context: &'value mut WriterContext,
+        config: ValueWriterConfig,
         raw_struct_writer: V::StructWriter,
     ) -> Self {
         Self {
             encoding: encoding_context,
             raw_struct_writer,
+            value_writer_config: config,
         }
     }
+
+    pub fn with_field_name_encoding(mut self, field_name_encoding: FieldNameEncoding) -> Self {
```

Comment:
🗺️ Notice that this lives on the struct writer, so you don't do 
```rust
writer.value_writer().with_field_name_encoding(E)
```
you do
```rust
writer.struct_writer().with_field_name_encoding(E)
```
which is a bit more obvious.